marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -177,10 +181,51 @@ handleAction AddNewWallet = do
     newWallets <- use _wallets
     liftEffect $ setItem walletsLocalStorageKey $ encodeJSON newWallets
 
--- contract actions
-handleAction (ContractAction contractAction) = do
-  case contractAction of
-    Contract.ClosePanel -> pure unit
-    action -> mapSubmodule _contractState ContractAction $ Contract.handleAction action
+-- template actions
+handleAction (SetTemplate template) = do
+  mCurrentTemplate <- peruse (_walletState <<< _templateState <<< _template)
+  case mCurrentTemplate of
+    Just currentTemplate
+      | currentTemplate == template -> pure unit
+    _ -> assign (_walletState <<< _templateState) $ Template.mkInitialState template
+  handleAction $ SetScreen $ ContractSetupScreen ContractRolesScreen
+
+handleAction (TemplateAction templateAction) = Template.handleAction templateAction
 
-handleAction (StartContract contract) = assign (_contractState <<< _executionState <<< _contract) contract
+-- contract actions
+handleAction StartContract = do
+  mTemplateState <- peruse (_walletState <<< _templateState)
+  for_ mTemplateState \templateState -> do
+    let
+      extendedContract = view (_template <<< _extendedContract) templateState
+    pure unit
+    let
+      templateContent = view _templateContent templateState
+    let
+      mContract = toCore $ fillTemplate templateContent extendedContract
```

Comment:
I think 198-205 can be written as:
```for_ mTemplateState \templateState ->
    let
      extendedContract = view (_template <<< _extendedContract) templateState
      templateContent = view _templateContent templateState
      mContract = toCore $ fillTemplate templateContent extendedContract
    in
```
Without `do` or `pure unit`

Reply:
Oops, don't know how that `pure unit` got in there. Must have been during a rewrite when I wasn't paying full attention. Thanks. 👍